File: 0387-first-unique-character-in-a-string/0387-first-unique-character-in-a-string.py
class Solution(object):
    def firstUniqChar(self, s):
        """
        :type s: str
        :rtype: int
        """
        n=len(s)
        mpp={}
        for i in range (n):
            mpp[s[i]]=mpp.get(s[i],0)+1
        for i in range(n):
            if mpp[s[i]]==1:
                return i
        return -1

File: 0387-first-unique-character-in-a-string/0387-first-unique-character-in-a-string.cpp
class Solution {
public:
    int firstUniqChar(string s) {
        int n=s.size();
        unordered_map<char,int> st;
        for(int i=0;i<n;i++){
            st[s[i]]++;
        }
        for(int i=0;i<n;i++){
            if(st[s[i]]==1) return i;
        }
        return -1;
    }
};